opm/utility/ECLGraph.cpp

```diff
@@ -1855,6 +1940,69 @@ flux(const PhaseIndex phase) const
     return v;
 }
 
+namespace Opm {
+
+    template <typename T>
+    std::vector<T>
+    ECLGraph::Impl::rawLinearisedCellData(const std::string& vector) const
+    {
+        auto x = std::vector<T>{};  x.reserve(this->numCells());
+
+        for (const auto& G : this->grid_) {
+            const auto xi =
+                G.template activeCellData<T>(*this->src_, vector);
+
+            x.insert(x.end(), std::begin(xi), std::end(xi));
+        }
+
+        if (x.size() != this->numCells()) {
+            return {};
+        }
+
+        return x;
+    }
+} // namespace Opm
+
+std::vector<double>
+Opm::ECLGraph::Impl::linearisedCellData(const std::string& vector,
+                                        UnitConvention     unit) const
+{
+    auto x = std::vector<double>{};  x.reserve(this->numCells());
+
+    auto gridID = 0;
```

Comment:
Since the grid ID is already stored in `class CartesianGridData` we should add a public query for it, that would avoid the slightly fragile logic around grid IDs in this function (and possible elsewhere, eventually).